modules/calib3d/src/ptsetreg.cpp

```diff
@@ -501,6 +501,269 @@ class Affine3DEstimatorCallback : public PointSetRegistrator::Callback
     }
 };
 
+class Affine2DEstimatorCallback : public PointSetRegistrator::Callback
+{
+public:
+    int runKernel( InputArray _m1, InputArray _m2, OutputArray _model ) const
+    {
+        Mat m1 = _m1.getMat(), m2 = _m2.getMat();
+        const Point2f* from = m1.ptr<Point2f>();
+        const Point2f* to   = m2.ptr<Point2f>();
+
+        // we need 3 points to estimate affine transform
+        const int N = 6;
+        double buf[N*N + N + N];
+        Mat A(N, N, CV_64F, &buf[0]);
+        Mat B(N, 1, CV_64F, &buf[0] + N*N);
+        Mat X(N, 1, CV_64F, &buf[0] + N*N + N);
+        double* Adata = A.ptr<double>();
+        double* Bdata = B.ptr<double>();
+        A = Scalar::all(0);
+
+        for( int i = 0; i < (N/2); i++ )
```

Comment:
Yes, I think you are right.  I did some tests on findHomography() to see what effect the final call to `cb->runKernel` has (it is run on all of the consensus points before starting the LM refinement.  Basically, if you have *either* that final call to `cb->runKernel` *or* the LM step you get the same results as if you have both.  My tests were with fairly stable transforms (adjacent video frames), but perhaps this additional step helps with stability on more extreme transitions.  I think your approach is fine.